src/libraries/Microsoft.Extensions.FileProviders.Physical/src/PhysicalFileProvider.cs

```diff
@@ -181,7 +181,6 @@ private void ReadPollingEnvironmentVariables()
         public void Dispose()
         {
             Dispose(true);
-            GC.SuppressFinalize(this);
```

Comment:
You should still leave `GC.SuppressFinalize(this);` even though the class doesn't have a finalizer. That way if a derived class does add a finalizer, it is still suppressed when disposing directly.

Reply:
Isn’t it usually the contract of the class that has the finalizer to suppress its own finalization?  Or are you suggesting it’s the responsibility of the class that exposes Dispose?  I see tons of examples in the framework where we don’t do this.

I could see how we should not remove this if it was already present but I’m not so sure about adding it.

Reply:
Reading https://docs.microsoft.com/en-us/dotnet/standard/garbage-collection/implementing-dispose#the-dispose-method, it indicates to me that the recommended pattern is to include it in the base class that exposes `Dispose()`.

> to indicate that the finalizer, if one is present, doesn't have to run.

and 

> Therefore, the call to the SuppressFinalize method prevents the garbage collector from running the finalizer. If the type has no finalizer, the call to GC.SuppressFinalize has no effect.

And later, when discussing how to [implement a derived class](https://docs.microsoft.com/en-us/dotnet/standard/garbage-collection/implementing-dispose#implement-the-dispose-pattern-for-a-derived-class), it makes no mention of calling `GC.SuppressFinalize`. It would be odd to tell derived classes that they would need to look at the base class to see if it calls `GC.SuppressFinalize`, and then call it yourself, if the base doesn't.

Reply:
> odd to tell derived classes that they would need to look at the base class 

Maybe.  The same docs tell you to consider that the base class may already define a finalizer and you don't need to define one.

In order to fix the product to be consistent with the docs we then need to add `GC.SuppressFinalize` to every class that exposes `protected Dispose(bool)`.  Even those without finalizers, despite are recommendation to avoid/minimize the use of finalizer.  Would you agree with that @eerhardt?

Reply:
Looking at some of the well-known base disposable classes, they all call SupressFinalize without having a finalizer. Stream and TextWriter both do.
I think we should follow our documented recommendations by default. If we think the docs should be updated, we should update them.

Reply:
> some of the well-known base disposable classes, they all call SupressFinalize

*some+all* is not the same as *all*.  I think it needs to be *all* for the docs to be correct and to make this recommendation on new code.  I've opened https://github.com/dotnet/runtime/issues/42183 to continue the discussion.